pkg/client/cache/store.go

```diff
@@ -153,6 +192,16 @@ func (c *cache) Update(obj interface{}) error {
 	c.lock.Lock()
 	defer c.lock.Unlock()
 	oldObject := c.items[key]
+
+	// reject attempts to add older entries
+	if older, err := c.isOlder(obj, oldObject); err != nil {
+		return fmt.Errorf("couldn't compare objects: %v", err)
```

Comment:
This structure is ugly, do the more canonical

older, err := c.IsOlder()
if err {}